CRM/Contact/BAO/Query.php

```diff
@@ -5829,12 +5811,40 @@ public function convertToPseudoNames(&$dao, $return = FALSE, $usedForAPI = FALSE
         $val = $dao->{$value['idCol']};
         if ($key == 'groups') {
           $dao->groups = $this->convertGroupIDStringToLabelString($dao, $val);
-          return;
+          continue;
```

Comment:
True didn't notice the foreach but have checked and seems ok but just something to watch out